python-backend/app/api/services/idir_service.py

```diff
@@ -0,0 +1,53 @@
+from ldap3 import Server, Connection, ObjectDef, Reader, ALL, NTLM
+from flask import current_app
+
+IDIR_URL = 'plywood.idir.bcgov'
+EMPR_DN = "OU=Users,OU=Energy Mines and Petroleum Resources,OU=BCGOV,DC=idir,DC=BCGOV"
+
+idir_field_map = {
+    "bcgov_guid": "bcgovGUID",
+    "username": "msDS-PrincipalName",
+    "email": "mail",
+    "phone_no": "telephoneNumber",
+    "title": "title",
+    "city": "l",
+    "department": "department"
+}
+
+
+class IdirService():
+    def search_for_users(basedn):
+        server = Server(IDIR_URL, get_info=ALL)
+        conn = Connection(
+            server, user="IDIR\\CSI_MMPO", password="CPqp1115", authentication=NTLM, auto_bind=True)
```

Comment:
These should probably be stored as openshift secrets and not in code.

Reply:
Done, just running one final test.

Reply:
and changed the password for anyone wondering.